python3Packages.proton-vpn-network-manager: 0.5.2 -> 0.6.3

Steinhagen opened the pull request:
## Description of changes

Update the package to the latest version.

The proton-vpn-network-manager-wireguard and proton-vpn-network-manager-openvpn modules are now legacy: The same functionality is now in proton-vpn-network-manager module, version 0.6.3 and upwards.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SebTM commented:
I think we have to split this in multiple commits for version-upgrade and replacing the two packages that are obsolete now with this one ✌🏻

SebTM commented:
Also I don't think these modules are used anywhere else in nixpkgs, so we might should drop them in another commit/pr, WDYT?

SebTM commented:
Result of `nixpkgs-review pr 343208` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>protonvpn-gui</li>
    <li>protonvpn-gui.dist</li>
    <li>python311Packages.proton-vpn-network-manager (python311Packages.proton-vpn-network-manager-openvpn ,python311Packages.proton-vpn-network-manager-wireguard)</li>
    <li>python311Packages.proton-vpn-network-manager.dist (python311Packages.proton-vpn-network-manager-openvpn.dist)</li>
    <li>python312Packages.proton-vpn-network-manager (python312Packages.proton-vpn-network-manager-openvpn ,python312Packages.proton-vpn-network-manager-wireguard)</li>
    <li>python312Packages.proton-vpn-network-manager.dist (python312Packages.proton-vpn-network-manager-openvpn.dist)</li>
  </ul>
</details>

Steinhagen committed:
python3Packages.proton-vpn-network-manager: 0.5.2 -> 0.6.3

Update to the latest versions.

The proton-vpn-network-manager-wireguard and proton-vpn-network-manager-openvpn
modules are now legacy: The same functionality is now in proton-vpn-network-manager module,
version 0.6.3 and upwards.

Steinhagen committed:
python3Packages.proton-vpn-network-manager: deprecates proton-vpn-network-manager-openvpn and proton-vpn-network-manager-wireguard

The proton-vpn-network-manager-wireguard and proton-vpn-network-manager-openvpn
modules are now legacy: The same functionality is now in proton-vpn-network-manager
module, version 0.6.3 and upwards.

This commit removes the 2 deprecated modules and creates a throw message for
them to the recommanded package to be used.

Steinhagen commented:
I've split the package update (and the removal of the other 2 packages from protonvpn-app) in one commit since the package covers everything now.
The second commit removes the deprecated packages and creates an alias for them to the recommended package to be used from now on.

Steinhagen commented:
> Also I don't think these modules are used anywhere else in nixpkgs, so we might should drop them in another commit/pr, WDYT?

Technically I'm for it. The only issue is if someone were those 2 packages, we will break their setup during the update with no warning.

SebTM commented:
Result of `nixpkgs-review pr 343208` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>protonvpn-gui</li>
    <li>protonvpn-gui.dist</li>
    <li>python311Packages.proton-vpn-network-manager (python311Packages.proton-vpn-network-manager-openvpn ,python311Packages.proton-vpn-network-manager-wireguard)</li>
    <li>python311Packages.proton-vpn-network-manager.dist (python311Packages.proton-vpn-network-manager-openvpn.dist)</li>
    <li>python312Packages.proton-vpn-network-manager (python312Packages.proton-vpn-network-manager-openvpn ,python312Packages.proton-vpn-network-manager-wireguard)</li>
    <li>python312Packages.proton-vpn-network-manager.dist (python312Packages.proton-vpn-network-manager-openvpn.dist)</li>
  </ul>
</details>

natsukium commented:
@ofborg build python312Packages.proton-vpn-network-manager

natsukium commented on pkgs/top-level/python-packages.nix:
```diff
@@ -10636,9 +10636,9 @@ self: super: with self; {
 
-  proton-vpn-network-manager-openvpn = callPackage ../development/python-modules/proton-vpn-network-manager-openvpn { };
+  proton-vpn-network-manager-openvpn = self.proton-vpn-network-manager; # Alias added 20-09-2024
 
-  proton-vpn-network-manager-wireguard = callPackage ../development/python-modules/proton-vpn-network-manager-wireguard { };
+  proton-vpn-network-manager-wireguard = self.proton-vpn-network-manager; # Alias added 20-09-2024
```
Please use `python-aliases.nix` instead.

Steinhagen replied:
I've converted the aliases into a throw error in python-aliases.nix since this is probably the better workflow in the long run.

SebTM commented:
Result of `nixpkgs-review pr 343208` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>protonvpn-gui</li>
    <li>protonvpn-gui.dist</li>
    <li>python311Packages.proton-vpn-network-manager</li>
    <li>python311Packages.proton-vpn-network-manager.dist</li>
    <li>python312Packages.proton-vpn-network-manager</li>
    <li>python312Packages.proton-vpn-network-manager.dist</li>
  </ul>
</details>

natsukium commented on pkgs/top-level/python-aliases.nix:
```diff
@@ -395,6 +395,8 @@ mapAliases ({
   prometheus_client = prometheus-client; # added 2021-06-10
   prompt_toolkit = prompt-toolkit; # added 2021-07-22
   protonup = protonup-ng; # Added 2022-11-06
+  proton-vpn-network-manager-openvpn = throw "proton-vpn-network-manager-openvpn functionality was integrated in the proton-vpn-network-manager module"; # added 20-09-2024
+  proton-vpn-network-manager-wireguard = throw "proton-vpn-network-manager-wireguard functionality was integrated in the proton-vpn-network-manager module"; # added 20-09-2024
```
```suggestion
  proton-vpn-network-manager-openvpn = throw "proton-vpn-network-manager-openvpn functionality was integrated in the proton-vpn-network-manager module"; # added 2024-09-20
  proton-vpn-network-manager-wireguard = throw "proton-vpn-network-manager-wireguard functionality was integrated in the proton-vpn-network-manager module"; # added 2024-09-20
```
The date format should be YYYY-MM-DD to clean up aliases with a script.
https://github.com/NixOS/nixpkgs/blob/9c711566cde5929768e311413eaa2399631624ce/maintainers/scripts/remove-old-aliases.py#L56-L61

Steinhagen replied:
Done.
